### PR TITLE
More feature flag deletions

### PR DESCRIPTION
**Pull request by terencechain**
Part of #7509 

This PR removes the following flags. The rationales are outlined in the issue above.
- `disable-broadcast-slashings`
- `disable-dynamic-committee-subnets`
- `disable-update-head-attestation`
- `disable-noise`
- `wait-for-synced`
- `disable-new-beacon-state-locks`
- `init-sync-verbose`
- `disable-finalized-deposits-cache`

Note: deep source failed coverage reported lines that's out of this scope

### CODE REVIEW

File: shared/featureconfig/flags.go [hunk 1]
```diff
@@ -167,33 +133,23 @@ var SlasherFlags = append(deprecatedFlags, []cli.Flag{
...
 	AltonaTestnet,
 	OnyxTestnet,
 	MedallaTestnet,
 	SpadinaTestnet,
 	ZinkenTestnet,
-	initSyncVerbose,
-	disableFinalizedDepositsCache,
 	enableBlst,
 	enableEth1DataMajorityVote,
-	enableAttBroadcastDiscoveryAttempts,
```
Comment (prestonvanloon) [inline shared/featureconfig/flags.go:196]: Are you removing this?

Comment (terencechain) [reply shared/featureconfig/flags.go:196]: My bad. I'm going to leave this in.

File: validator/client/runner.go [hunk 2]
```diff
@@ -62,18 +62,14 @@ func run(ctx context.Context, v Validator) {
 			log.Fatalf("Slasher is not ready: %v", err)
 		}
 	}
-	if featureconfig.Get().WaitForSynced {
-		if err := v.WaitForSynced(ctx); err != nil {
```
Comment (prestonvanloon) [inline validator/client/runner.go:66]: Are you going to delete this method too?

Comment (prestonvanloon) [reply validator/client/runner.go:66]: This feature looks like it wasn't launched yet, but you are removing it?

Comment (prestonvanloon) [reply validator/client/runner.go:66]: cc: @0xKiwi

Comment (terencechain) [reply validator/client/runner.go:66]: Similar to `StateGenSigVerify`, this looks to be a feature that we should never release on beta/mainnet. This skips WaitForChainStart and is only utilized during test (e.g. e2e)

Comment (terencechain) [reply validator/client/runner.go:66]: Chatted with @0xKiwi this feature is not done. We'll leave it in for now

File: beacon-chain/sync/metrics.go [hunk 3]
```diff
@@ -94,17 +92,11 @@ func (s *Service) updateMetrics() {
...
-		}
-	} else {
-		for _, committeeIdx := range indices {
-			formattedTopic := fmt.Sprintf(attTopic, digest, committeeIdx)
-			topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.p2p.PubSub().ListPeers(formattedTopic))))
-		}
+	for _, committeeIdx := range indices {
+		formattedTopic := fmt.Sprintf(attTopic, digest, committeeIdx)
+		topicPeerCount.WithLabelValues(formattedTopic).Set(float64(len(s.p2p.PubSub().ListPeers(formattedTopic))))
 	}
```
Comment (prestonvanloon) [inline beacon-chain/sync/metrics.go:98]: I don't think this is a good idea. I think it should always be in the path of DisableDynamicCommitteeSubnets. 
If was on committee 5 and I have 10 peers on that subnet. Prometheus polls this metric, and the topic is updated with the value 10. 

Now, one epoch later, I am no longer on committee 5. When prometheus polls the metrics again, the value for committee subnet 5 is still 10 and it is not updated. This leads to stale and inaccurate results.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**